### PR TITLE
validate product has no variant if adding master

### PR DESCRIPTION
**Pull request by nimish-mehta**

### CODE REVIEW

File: apps/ex_shop/test/models/line_item_test.exs [hunk 1]
```diff
@@ -21,10 +21,28 @@ defmodule ExShop.LineItemTest do
...
+  @product_attr Map.merge(@product_data, @product_master_variant_data) |> Map.merge(@variant_option_value_attrs)
+
+  @valid_variant_attrs %{
+    cost_price: "120.5",
+    discontinue_on: %{"year" => "2016", "month" => "2", "day" => "1"},
+    height: "120.5", weight: "120.5", width: "120.5",
+    sku: "URG123"
+  }
 
-  @tag :pending
```
Comment (pikender) [inline apps/ex_shop/test/models/line_item_test.exs:26]: please keep the pending test as we have to add more tests where we will re-check all the tests with non-master variants :)

File: apps/ex_shop/test/models/line_item_test.exs [hunk 2]
```diff
@@ -82,10 +100,27 @@ defmodule ExShop.LineItemTest do
     product.master
   end
 
+  defp create_product_with_variant do
+    master_variant = create_product
+    product = master_variant |> Repo.preload([:product]) |> Map.get(:product)
```
Comment (pikender) [inline apps/ex_shop/test/models/line_item_test.exs]: Can we please avoid by refactoring create_product (I skipped it :( ) to provide created product directly ??

or Add a TODO for now, need to revisit

File: apps/ex_shop/web/models/line_item.ex [hunk 3]
```diff
@@ -112,4 +114,16 @@ defmodule ExShop.LineItem do
     end
   end
 
+  defp ensure_product_has_no_variants_if_master(changeset) do
+    variant =
+      changeset.model
+      |> Repo.preload([variant: :product])
+      |> Map.get(:variant)
+    if variant.is_master and Product.variant_count(variant.product) > 1 do
```
Comment (pikender) [inline apps/ex_shop/web/models/line_item.ex]: Can we add function product_has_variants in Product instead and re-use it ??

File: apps/ex_shop/test/models/line_item_test.exs (hunk 2 above)
Comment (nimish-mehta) [inline apps/ex_shop/test/models/line_item_test.exs]: refactored

**Pull request merged**